File: tournaments/factorialsProductTrailingZeros/factorialsProductTrailingZeros.cpp
int factorialsProductTrailingZeros(int l, int r) {
  int result = 0,
      last = 0;
  for (int i = 1; i <= r; i++) {
    int number = i;
    while (number % 5 == 0) {
      number /= 5;
      last++;
    }
    if (i >= l) {
      result += last;
    }
  }
  return result;
}

File: tournaments/factorialsProductTrailingZeros/factorialsProductTrailingZeros.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub factorialsProductTrailingZeros {
    my ($l, $r) = @_;

    my $result = 0;
    my $index = 0;

    for (my $i = 1; $r >= $i; ++$i) {
        my $temp = $i;

        until ($temp % 5) {
            $temp /= 5;
            ++$index;
        }
        if ($l <= $i) {
            $result += $index;
        }
    }

    return $result;
}
